### PR TITLE
EKS cluster with cluster-controller and without public API access

### PR DESCRIPTION
**Pull request by matas-cast**
https://castai.atlassian.net/browse/CORE-2084

### CODE REVIEW

File: examples/eks_private_cluster/main.tf [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    value = "eks"
+  }
+
+  set_sensitive {
+    name  = "apiKey"
+    value = castai_eks_cluster.my_castai_cluster.cluster_token
+  }
+}
+
+resource "castai_eks_cluster" "my_castai_cluster" {
```
Comment (anjmao) [inline examples/eks_private_cluster/main.tf:57]: Maybe put castai_eks_cluster resource before castai_agent helm install

File: examples/eks_private_cluster/vpc.tf [hunk 2]
```diff
@@ -0,0 +1,81 @@
+#1. create VPC
+
+data "aws_availability_zones" "available" {}
+
+### VPC
```
Comment (anjmao) [inline examples/eks_private_cluster/vpc.tf:5]: `### VPC` comment is not very useful, maybe remove it

**Review by anjmao — APPROVED** [review]

File: examples/eks_private_cluster/README.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+## EKS and CAST.AI setup example
+
+Reference setup for EKS clusters without public API access, connected to CAST.AI
```
Comment (anjmao) [inline examples/eks_private_cluster/README.md:3]: Replace `CAST.AI` -> `CAST AI` in all places

Comment (anjmao) [inline examples/eks_private_cluster/README.md:3]: Maybe simpler `Reference setup for EKS clusters connected to CAST.AI`

There is no need to mention network topology as we don't care if we have api access.

File: examples/eks_private_cluster/README.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+
+### Instruction for full setup (create vpc, eks and connect with cast.ai)
```
Comment (anjmao) [inline examples/eks_private_cluster/README.md:5]: cast.ai -> CAST AI

File: examples/eks_private_cluster/eks.tf [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+      rolearn  = module.castai-eks-iam.instance_profile_role_arn
+      username = "system:node:{{EC2PrivateDNSName}}"
+      groups   = ["system:bootstrappers", "system:nodes"]
+    },
+  ]
+}
+
+data "aws_eks_cluster" "eks" {
+  name = module.eks.cluster_id
+}
```
Comment (anjmao) [inline examples/eks_private_cluster/eks.tf:61]: Add new line

**Review by anjmao — APPROVED** [review]

File: examples/eks_private_cluster/README.md (hunk 3 above)
Comment (matas-cast) [reply examples/eks_private_cluster/README.md]: thats true. initially we had a separate example for eks without controller. I think I'll do that with next PR though

**Pull request merged**